tests/index.test.js

```diff
@@ -0,0 +1,108 @@
+import nock from 'nock';
+
+import fetchRedditPostsByTopic, { fetchJson, extractFields, getRedditUrl } from '../lib/index.js';
+
+describe('fetchRedditPostsByTopicPromise', () => {
+
+  describe('Fetch json', () => {
+
+    const response = { 
+      data: { 
+        children: [
+          { data: { id: 1, title: 'RTX 3080 Crashes', thumbnail: 'self', subreddit: 'Nvidia', ups: 635, downs: 34 } },
+          { data: { id: 2, title: 'Cyberpunk 2077 goes gold', thumbnail: 'self', subreddit: 'Nvidia', ups: 1124, downs: 450 } },
+          { data: { id: 3, title: 'PS5 release data', thumbnail: 'self', subreddit: 'Nvidia', ups: 12903 , downs: 2415 } }
+        ] 
+      } 
+    };
+
+    it('Should match returned data', async () => {
+      const redditUrl = 'https://reddit.com/r/pics/search.json?q=Gaming&sort=new';
+
+      const responseData = { data: { children: [{ title: 'RTX 3080 Crashes' }] } };
+
+      nock('https://reddit.com')
+        .get('/r/pics/search.json?q=Gaming&sort=new')
+        .reply(200, responseData);
+
+      return await fetchJson(redditUrl)
```

Comment:
Let's clean up this syntax. Currently it's a mix of async/await and a .then promise chain.

Let's just use async/await similar to test below.